shared/src/error/mod.rs

```diff
@@ -114,6 +114,7 @@ impl ExitCode {
 /// When a syscall fails, it returns an `ErrorNumber` to indicate why. The syscalls themselves
 /// include documentation on _which_ syscall errors they can be expected to return, and what they
 /// mean in the context of the syscall.
+#[non_exhaustive]
```

Comment:
We don't _have_ to do this, but it's easier to make this "extensible" now than it is to do that later.

Reply:
Removing the `non_exhaustive` later would just mean that users can now exhaustively match, it won't affect existing code.